Description: add support for the ubuntu lock on suspend option 
Author: Tim Lunn <tim@feathertop.org>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1063110

---
---
 js/ui/screenShield.js |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: gnome-shell-3.28.1/js/ui/screenShield.js
===================================================================
--- gnome-shell-3.28.1.orig/js/ui/screenShield.js
+++ gnome-shell-3.28.1/js/ui/screenShield.js
@@ -32,6 +32,7 @@ const Util = imports.misc.util;
 const SCREENSAVER_SCHEMA = 'org.gnome.desktop.screensaver';
 const LOCK_ENABLED_KEY = 'lock-enabled';
 const LOCK_DELAY_KEY = 'lock-delay';
+const SUSPEND_LOCK_ENABLED_KEY = 'ubuntu-lock-on-suspend';
 
 const LOCKDOWN_SCHEMA = 'org.gnome.desktop.lockdown';
 const DISABLE_LOCK_KEY = 'disable-lock-screen';
@@ -543,6 +544,7 @@ var ScreenShield = new Lang.Class({
 
         this._settings = new Gio.Settings({ schema_id: SCREENSAVER_SCHEMA });
         this._settings.connect('changed::' + LOCK_ENABLED_KEY, this._syncInhibitor.bind(this));
+        this._settings.connect('changed::' + SUSPEND_LOCK_ENABLED_KEY, this._syncInhibitor.bind(this));
 
         this._lockSettings = new Gio.Settings({ schema_id: LOCKDOWN_SCHEMA });
         this._lockSettings.connect('changed::' + DISABLE_LOCK_KEY, this._syncInhibitor.bind(this));
@@ -707,7 +709,8 @@ var ScreenShield = new Lang.Class({
     },
 
     _syncInhibitor() {
-        let lockEnabled = this._settings.get_boolean(LOCK_ENABLED_KEY);
+        let lockEnabled = this._settings.get_boolean(LOCK_ENABLED_KEY) ||
+                          this._settings.get_boolean(SUSPEND_LOCK_ENABLED_KEY);
         let lockLocked = this._lockSettings.get_boolean(DISABLE_LOCK_KEY);
         let inhibit = (this._loginSession && this._loginSession.Active &&
                        !this._isActive && lockEnabled && !lockLocked);
@@ -727,7 +730,7 @@ var ScreenShield = new Lang.Class({
 
     _prepareForSleep(loginManager, aboutToSuspend) {
         if (aboutToSuspend) {
-            if (this._settings.get_boolean(LOCK_ENABLED_KEY))
+            if (this._settings.get_boolean(SUSPEND_LOCK_ENABLED_KEY))
                 this.lock(true);
         } else {
             this._wakeUpScreen();
